universum/modules/output/html_output.py

```diff
@@ -110,47 +107,49 @@ def _log_buffered(self, line):
             self._log_and_clear_buffer()
         self._write_to_file(line)
 
-    def _log_and_clear_buffer(self):
-        for buffered_line in self._log_buffer:
+    def _log_and_clear_buffer(self) -> None:
+        for buffered_line in self._log_buffer:  # type: str
             self._write_to_file(buffered_line)
         self._log_buffer = []
 
-    def _write_to_file(self, line):
+    def _write_to_file(self, line: str) -> None:
+        if not self._log_path:
+            raise RuntimeError("Artifact directory was not set")
+
         with open(self._log_path, "a", encoding="utf-8") as file:
             file.write(line)
 
-    def _build_indent(self):
-        indent_str = []
-        for x in range(0, self._block_level):
+    def _build_indent(self) -> str:
+        indent_str: List[str] = []
+        for x in range(0, self._block_level):  # type: int
```

Comment:
do you really need to specify type here? I'd assume that mypy figures it out based on range() return